### ISSUE TITLE
Go coordinate support

### ISSUE THREAD

**Issue by jeffmcaffer** [clearlydefined/service#765]
Go is of course very popular and it would be great to have support for tracking Go components. As the Go module system has evolved, things have gotten clearer and easier to implement for ClearlyDefined. However, there is still the challenge of how to identify Go components using ClearlyDefined `coordinates`. Typical package management systems have a small number of registries (providers in ClearlyDefined-speak). The current approach has been to know these up front and code them into the services. This allows the team to effectively curate the sources from which we are pulling data and account for any slight variations in authentication, rate limiting, etc. 

With Go, while many of the modules are in Git repos on GitHub, that is by no means the universal case. We need to balance a more flexible coordinate structure for Go packages with the realities of accessing arbitrary content on the internet. With some deeper knowledge of the Go modules infrastructure we can likely figure something out and implement.

**Comment by fossygirl** [comment]
@jeffmendoza I believe you did some investigation into Go support in ClearlyDefined. Could you summarize your conclusions here?

**Comment by jeffmendoza** [comment]
Here is my thinking from before:
https://docs.google.com/document/d/1n75c9rVbGj95NDROrkbNovB9FTo6g0jTE-zhKREDbyY/edit?usp=sharing
I think "option B" is the right way to go there.

**Comment by sschuberth** [comment]
With Go support being merged in #871, can this be closed?

**Comment by capfei** [comment]
@nellshamrell Can we close this one? We can track Go support prior to 1.11 in a separate issue, if we want to add that.

**Comment by nellshamrell** [comment]
Yes, this can definitely be closed. Closing :)
